cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java

```diff
@@ -330,6 +328,41 @@ private ConfiguredStage configureStage(ETLStage stage, ValidatedPipeline validat
     return specBuilder;
   }
 
+  private Object getPlugin(String stageName, ETLPlugin etlPlugin, PluginSelector pluginSelector, String type,
+                           String pluginName, DefaultStageConfigurer stageConfigurer) {
+    Object plugin = null;
+    try {
+      plugin = pluginConfigurer.usePlugin(type, pluginName, stageName, etlPlugin.getPluginProperties(), pluginSelector);
+    } catch (IllegalArgumentException e) {
+      stageConfigurer.addFailure(e.getMessage(), String.format("Provide unique plugin id for plugin '%s'.", stageName));
+    } catch (InvalidPluginConfigException e) {
+      stageConfigurer.addFailure(e.getMessage(), "Provide valid plugin config properties.");
+    }
+    // throw validation exception if any error occurred while creating a new instance of the plugin
+    stageConfigurer.throwIfFailure();
+
+    if (plugin == null) {
+      String errorMessage = String.format("Plugin named '%s' of type '%s' not found.", pluginName, type);
+      String correctiveAction = String.format("Make sure plugin '%s' of type '%s' is already deployed.",
+                                              pluginName, type);
+      stageConfigurer.addFailure(errorMessage, correctiveAction).withPluginNotFoundCause(stageName, pluginName, type);
```

Comment:
this validation error contains less information than it used to (information about a suggested artifact). We should still include that information. This was done specifically for pipeline import, to remove a lot of the logic that currently resides in the UI.